pkg/typesystem/typesystem.go

```diff
@@ -165,13 +168,31 @@ type TypeSystem struct {
 	// [objectType] => [relationName] => TTU relation.
 	ttuRelations map[string]map[string][]*openfgav1.TupleToUserset
 
-	modelID       string
-	schemaVersion string
+	modelID                 string
+	schemaVersion           string
+	authorizationModelGraph *graph.AuthorizationModelGraph
+	authorizationModelNodes map[string]*graph.AuthorizationModelNode
+}
+
+// getAllAuthorizationModelNodes is a helper function to get all authorization model nodes from the authorization model graph.
+func getAllAuthorizationModelNodes(g *multi.DirectedGraph) (map[string]*graph.AuthorizationModelNode, error) {
+	authorizationModelNodes := make(map[string]*graph.AuthorizationModelNode)
+	nodes := g.Nodes()
+	for nodes.Next() {
+		curNode := nodes.Node()
+		curAuthorizationModelNode, ok := curNode.(*graph.AuthorizationModelNode)
+		if !ok {
+			// Strange, I should have expected to be able to cast to AuthorizationModelNode.
+			return nil, fmt.Errorf("unable to cast as graph.AuthorizationModelNode. node ID: %v", curNode.ID())
+		}
+		authorizationModelNodes[curAuthorizationModelNode.Label()] = curAuthorizationModelNode
```

Comment:
**Thought**: i'm kind of regretting calling `.Label()` like that. It's not clear enough. I think we should change it, but i'm not sure what to call it. WDYT?